lib/spack/spack/solver/concretize.lp

```diff
@@ -1432,11 +1432,11 @@ opt_criterion(73, "deprecated versions used").
 % 1. Version weight
 % 2. Number of variants with a non default value, if not set
 % for the root package.
-opt_criterion(70, "version weight").
+opt_criterion(70, "version badness (roots)").
 #minimize{ 0@270: #true }.
 #minimize{ 0@70: #true }.
 #minimize {
-    Weight@70+Priority
+    Weight@70+Priority,PackageNode
```

Comment:
@haampie This is the bug. Minimization happens on a set of tuples, by summing the first item (that's how clingo works). Since we didn't add `PackageNode` to distinguish among two different roots, we were considering two penalties of the same value as the same item. 

Basically, in `develop` the minimization is on:
```
{ (2,) }
```
and now it is on:
```
{ (2, lcov), (2, vtk) }
```
Summing the first gives a penalty of 2, where it should have been 4 .

Reply:
> Why are roots special here?

Cause, apparently since ever, we forgot to add `PackageNode` to distinguish tuples. All other criteria have it.